packages/vue/src/pages/planning-task-list/components/task-list-grid.vue

```diff
@@ -0,0 +1,259 @@
+            <dx-text-box :readOnly="true"></dx-text-box>
+          </div>
+        </template>
+        <template #item="{ data }">
+          <task-priority :value="data" class="task-priority"></task-priority>
+        </template>
+      </dx-select-box>
+    </template>
+
+    <template #cellStatus="{ data }">
+      <div v-html="getStatusHtml(data.value)"/>
+    </template>
+
+    <template #editCellStatus="{ data: cellInfo }">
+      <dx-select-box
+        class="edit-cell"
+        :value="cellInfo.value"
+        :items="statusList"
+        @value-changed="(e) => {
+          cellInfo.setValue(e.value);
+          cellInfo.component.refresh();
+        }"
+        @selection-changed="cellInfo.component.updateDimensions"
+        fieldTemplate="field"
+      >
+        <template #field>
+          <div class="task-status" v-html="getStatusHtml(cellInfo.value || '')"/>
+          <dx-text-box :readOnly="true"></dx-text-box>
+        </template>
+        <template #item="{data}">
+          <div v-html="getStatusHtml(data)"/>
+        </template>
+      </dx-select-box>
+    </template>
+  </dx-data-grid>
+  <dx-load-panel
+    :visible="props.isLoading"
+    container=".content"
+    :position="{ of: '.content' }"
+  />
+</template>
+
+<script setup lang="ts">
+import { ref } from 'vue';
+import {
+  DxDataGrid,
+  DxColumn,
+  DxHeaderFilter,
+  DxLoadPanel,
+  DxRequiredRule,
+  DxSelection,
+  DxScrolling,
+  DxSorting,
+  DxColumnLookup,
+  DxPager,
+  DxPaging,
+  DxEditing,
+} from 'devextreme-vue/data-grid';
+import 'jspdf-autotable';
+import { DxTextBox } from 'devextreme-vue/text-box';
+import { DxSelectBox } from 'devextreme-vue/select-box';
+import { jsPDF as JsPDF } from 'jspdf';
+import { saveAs } from 'file-saver-es';
+import { exportDataGrid as exportGridToPdf } from 'devextreme/pdf_exporter';
+import { exportDataGrid as exportToXLSX } from 'devextreme/excel_exporter';
+import type DataSource from 'devextreme/data/data_source';
+
+import type { Task } from '@/types/task';
+import type { RowPreparedEvent } from 'devextreme/ui/data_grid';
+import { taskPriorityList as priorityList, taskStatusList as statusList } from '@/types/task';
+import { Workbook } from 'exceljs';
+import TaskPriority from './components/task-priority.vue';
+
+const props = withDefaults(defineProps<{
+  dataSource: DataSource | null
+}>(), {
+  dataSource: () => null,
+});
+
+const dxDataGridCmp = ref<InstanceType<typeof DxDataGrid> | null>(null);
+
+const onRowPreparedGrid = (e: RowPreparedEvent<Task, number>) => {
+  const { rowType, rowElement, data } = e;
+
+  if (rowType === 'header') return;
+
+  if (data.status === 'Completed') {
+    rowElement.classList.add('completed');
+  }
+};
+
+const addRow = () => dxDataGridCmp.value?.instance.addRow();
+const getStatusHtml = (status: string) => `<span
+            class="status status-${status?.replace(/ /g, '-').toLowerCase()}">${status}
```

Comment:
Looks like we need create status componen. Because we will use it in detail page